pkg/operation/botanist/resource_manager.go

```diff
@@ -66,3 +68,35 @@ func (b *Botanist) WaitUntilManagedResourcesDeleted(ctx context.Context) error {
 		return retry.MinorError(fmt.Errorf("not all managed resources have been deleted in the shoot cluster (still existing: %s)", names))
 	})
 }
+
+// CleanManagedResourcesFinalizers cleans all managed resources finalizers.
+func (b *Botanist) CleanManagedResourcesFinalizers(ctx context.Context) error {
+	managedResources, err := b.listManagedResources(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to list all managed resource, %v", err)
+	}
+
+	if len(managedResources.Items) == 0 {
+		return nil
+	}
+
+	for _, resource := range managedResources.Items {
+		if err := kutil.TryUpdate(ctx, k8sretry.DefaultRetry, b.K8sSeedClient.Client(), &resource, func() error {
+			resource.SetFinalizers([]string{})
```

Comment:
```suggestion
			resource.SetFinalizers(nil)
```

Reply:
Does it make sense to also teach the resource-manager do to a migration instead of silently patching the finalizers here? Should the `ManagedResources` also have a migrate operation annotation? The resource-manager could remove the finalizers itself and doesn't care anymore about these resources?

Reply:
We considered such scenario, but we had concerns that there might be an idea where resource manager might not be always used in the context of Gardener.

Reply:
Hm, what? Can you elaborate? Why not? GRM is a central component that is used by g/g and most extensions.

Reply:
My point was, can't someone deploy the resource manager without having Gardener and benefit from its functionality? If so should resource manager understand of specific Gardener lifecycle processes? It was more a question but I was not sure so I proposed it like this. If not I will open PR int the GRM to handle such scenario.